Assets/Mirage/Runtime/NetworkBehaviour.cs

```diff
@@ -159,18 +175,22 @@ public NetworkIdentity Identity
                 // instead of calling unity's MonoBehaviour == operator
                 if (_identity is null)
                 {
-                    // GetComponentInParent doesn't works on disabled gameObject
-                    // and GetComponentsInParent(false)[0] isn't allocation free, so
-                    // we just drop child support in this specific case
-                    if (gameObject.activeInHierarchy)
-                        _identity = GetComponentInParent<NetworkIdentity>();
-                    else
-                        _identity = GetComponent<NetworkIdentity>();
+                    // we look up NetworkIdentity on this GO or any of its parents;
+                    // this allows placing NetworkBehaviours on children of NetworkIdentity
+#if UNITY_2021_2_OR_NEWER
+                    _identity = GetComponentInParent<NetworkIdentity>(true);
+#elif UNITY_2020_1_OR_NEWER
+                    _identity = gameObject.GetComponentInParent<NetworkIdentity>(true);
+#else
+                    // TODO: remove this bit once Unity drops support for 2019 LTS
+                    GetComponentsInParent<NetworkIdentity>(true, networkIdentityGetComponentCacheList);
+                    _identity = networkIdentityGetComponentCacheList[0];
+#endif
```

Comment:
Can you move this code into a new private function too, called something like `TryFindIdentity`. Just so that it is easier to read with all the `#if` stuff in its own function rather than the property.

I think after that this should be good to merge

Reply:
Alright, I will update it a bit later today.

Reply:
Should I also include updated LINQ code to NIs `NetworkBehaviours`?

Reply:
> Should I also include updated LINQ code to NI's `NetworkBehaviours`?

no, we can sort that out in #970 or another PR

Reply:
> Can you move this code into a new private function too, called something like `TryFindIdentity`.

Done.